File: 0496-next-greater-element-i/0496-next-greater-element-i.java
class Solution {
    public int[] nextGreaterElement(int[] nums1, int[] nums2) {
        int j,k=0;
        for(int i=0;i<nums1.length;i++){
            for(j=0;j<nums2.length;j++){
                if(nums1[i]==nums2[j]){
                    for(k=j+1;k<nums2.length;k++){
                        if(nums1[i]<nums2[k]){
                            nums1[i]=nums2[k];
                            break;
                        }
                    }
                    break;
                }
            }
            if(k==nums2.length)
            {
            nums1[i]=-1;    
             }
        }
        return nums1;
    }
}

File: 0496-next-greater-element-i/0496-next-greater-element-i.kt
import java.util.ArrayDeque
class Solution {
    fun nextGreaterElement(nums1: IntArray, nums2: IntArray): IntArray {
        var res=IntArray(nums1.size){-1}
        var stack=ArrayDeque<Int>()
        var hm=HashMap<Int,Int>()
        
        for(i in 0 until nums1.size){
            hm.put(nums1[i],i)
        }
        
        for(i in 0 until nums2.size){
            if(stack.isEmpty()){
                stack.push(nums2[i])
            }else{
                while(stack.peek()!=null && stack.peek()<nums2[i]){
                    if(hm.containsKey(stack.peek())){
                        res[hm[stack.peek()]!!]=nums2[i]
                    }
                    stack.pop()
                }
                stack.push(nums2[i])    
            }
        }
        
        return res
    }
}